### PR TITLE
ssht: conan v2 support

### PR DESCRIPTION
**Pull request by holger-detering**
Specify library name and version:  **ssht/1.3.7**, **ssht/1.4.0**

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/ssht/all/conanfile.py [hunk 1]
```diff
@@ -13,47 +17,52 @@ class SshtConan(ConanFile):
...
+        tc.cache_variables["tests"] = False
+        tc.cache_variables["python"] = False
+        tc.generate()
+        tc = CMakeDeps(self)
+        tc.generate()
+
 
     def build(self):
-        self.cmake.build()
+        replace_in_file(self, os.path.join(self.source_folder, "src", "c", "CMakeLists.txt"),
```
Comment (jcar87) [inline recipes/ssht/all/conanfile.py:55]: I think we could do: 
https://docs.conan.io/2/reference/tools/cmake/cmakedeps.html#overwrite-properties-from-the-consumer-side-using-cmakedeps-set-property

in the `generate()` method to make the target appear as `FFTW3::FFTW3` and avoid patching the recipe

File: recipes/ssht/all/test_v1_package/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,9 @@
+cmake_minimum_required(VERSION 3.15)
```
Comment (jcar87) [inline recipes/ssht/all/test_v1_package/CMakeLists.txt:1]: we can probably remove the entire `test_v1_package` folder - if it did not exist prior to this PR, we are no longer require newly updated recipes to include it

**Comment by czoido** [comment]
Hi @holger-detering,
Thanks a lot for the contribution, the PR seems good and it's also passing ci, but I agree with @jcar87 comments. We will push those changes and mark the PR as ready for review unless you want to comment anything more on this.

**Review by jcar87 — APPROVED** [review]

File: recipes/ssht/all/conanfile.py [hunk 3]
```diff
@@ -1,9 +1,13 @@
 from conan import ConanFile
-from conan.tools import files
 from conan.errors import ConanInvalidConfiguration
-from conans import CMake
+from conan.tools.microsoft import is_msvc
+from conan.tools.files import get, copy, rmdir, replace_in_file
```
Comment (jcar87) [inline recipes/ssht/all/conanfile.py:4]:
```suggestion
from conan.tools.files import get, copy, rmdir
```

**Review by czoido — APPROVED** [review]

**Review by jcar87 — APPROVED** [review]

**Pull request merged**